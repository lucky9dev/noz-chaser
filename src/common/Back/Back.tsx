import React, { useCallback } from 'react';
import { useHistory } from 'react-router-dom';
import Button from 'semantic-ui-react/dist/commonjs/elements/Button';
import Icon from 'semantic-ui-react/dist/commonjs/elements/Icon';
import useStore from 'store';

const Back: React.FC = () => {
  const history = useHistory();
  const darkMode = useStore(useCallback((state) => state.darkMode, []));

  const handleBack = () => {
    history.goBack();
  };

  return (
    <Button onClick={handleBack} style={{ height: 36 }} inverted={darkMode} type="button">
      <Icon name="arrow left" />
      Back
    </Button>
  );
};

export default Back;
