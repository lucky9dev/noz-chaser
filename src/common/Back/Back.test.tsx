import React from 'react';
import { render } from '@testing-library/react';
import Back from './Back';
import '@testing-library/jest-dom';

describe('Back', () => {
  test('Back renders correctly', () => {
    const { container } = render(<Back />);
    expect(container).toMatchSnapshot();
  });
});
