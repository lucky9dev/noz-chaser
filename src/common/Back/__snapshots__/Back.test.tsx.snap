// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Back renders correctly 1`] = `
<div>
  <button
    class="ui button"
    style="height: 36px;"
    type="button"
  >
    <i
      aria-hidden="true"
      class="arrow left icon"
    />
    Back
  </button>
</div>
`;
